projects/packages/publicize/src/class-publicize.php

```diff
@@ -630,6 +631,16 @@ public function post_is_done_sharing( $post_id = null ) {
 		return 'publish' === $post->post_status || get_post_meta( $post->ID, $this->POST_DONE . 'all', true ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 	}
 
+	/**
+	 * Returns whether or not the nudge to show the upgrade to Jetpack Social Plan.
+	 *
+	 * @return bool True if the nudge has to be shown, false otherwise.
+	 */
+	public function should_show_social_plan_nudge() {
+		Current_Plan::refresh_from_wpcom();
```

Comment:
Is this cached? It should be ok, but that will be making another API request each time the initial state is rendered.

Reply:
The `Current_Plan::supports` function in the next line calls `Current_Plan:: get()` which is cached. 

However, if we associate new features, the cached plan won't get busted. Jeptack refreshes the plan on every heart beat, so that's once a day. But Jetpack [also makes a call to `update_from_sites_response`, where  the plan is also refreshed on every page refresh](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#:~:text=Jetpack_Plan%3A%3Aupdate_from_sites_response(%20%24response%20)%3B) within the Jetpack Plugin. 

So, I guess it's ok to make the API call every time we render initial state.

Reply:
FWIW, it only gets cached internally in the class, which means it won't get called multiple times on the same request. Each request will still result in a new call.

Can you link to the spot where Jetpack calls that endpoint or function on every page request? I'm curious because it does seem like a lot.

Reply:
This is the place where Jetpack calls it 

https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#:~:text=Jetpack_Plan%3A%3Aupdate_from_sites_response(%20%24response%20)%3B

Reply:
This is the place where Jetpack calls it on every page request https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#:~:text=Jetpack_Plan%3A%3Aupdate_from_sites_response(%20%24response%20)%3B